packages/store/src/Storage.sol

```diff
@@ -113,69 +105,77 @@ library Storage {
     }
   }
 
-  function load(uint256 storagePointer, uint256 length) internal view returns (bytes memory) {
-    return load(storagePointer, length, 0);
-  }
-
   /**
-   * @notice Load raw bytes from storage at the given storagePointer, offset, and length
+   * Load raw bytes from storage at the given storagePointer, offset, and length
    */
   function load(uint256 storagePointer, uint256 length, uint256 offset) internal view returns (bytes memory result) {
-    // TODO this will probably use less gas via manual memory allocation
-    // (see https://github.com/latticexyz/mud/issues/444)
-    result = new bytes(length);
     uint256 memoryPointer;
+    /// @solidity memory-safe-assembly
     assembly {
+      // Solidity's YulUtilFunctions::roundUpFunction
+      function round_up_to_mul_of_32(value) -> _result {
+        _result := and(add(value, 31), not(31))
+      }
```

Comment:
can't really make a solidity function for this so figured I'd just copy the one solidity uses internally